src/vaadin-tooltip.html

```diff
@@ -106,13 +121,37 @@
           const targetLeft = targetRect.left - parentRect.left;
           const targetTop = targetRect.top - parentRect.top;
           const horizontalCenterOffset = (targetRect.width - thisRect.width) / 2;
-          const tooltipLeft = targetLeft + horizontalCenterOffset;
-          const tooltipTop = targetTop - thisRect.height;
+          const verticalCenterOffset = (targetRect.height - thisRect.height) / 2;
+          let tooltipLeft, tooltipTop;
+
+          switch (this.position) {
+            case 'top':
+              tooltipLeft = targetLeft + horizontalCenterOffset;
+              tooltipTop = targetTop - thisRect.height;
+              break;
+            case 'bottom':
+              tooltipLeft = targetLeft + horizontalCenterOffset;
+              tooltipTop = targetTop + targetRect.height;
+              break;
+            case 'left':
+              tooltipLeft = targetLeft - thisRect.width;
+              tooltipTop = targetTop + verticalCenterOffset;
+              break;
+            case 'right':
+              tooltipLeft = targetLeft + targetRect.width;
+              tooltipTop = targetTop + verticalCenterOffset;
+              break;
+          }
 
           this.style.top = `${tooltipTop}px`;
           this.style.left = `${tooltipLeft}px`;
         }
 
+        show() {
+          this.$.tooltip.classList.remove('hidden');
```

Comment:
Pls introduce a tooltipHidden property in component's props and then set `this.tooltipHidden = false` instead of doing this. Also it requires to add `display: none` to `:host([tooltip-hidden]) #tooltip` to CSS.